specification/mongocluster/DocumentDB.MongoCluster.Management/client.tsp

```diff
@@ -2,5 +2,17 @@ import "./main.tsp";
 import "@azure-tools/typespec-client-generator-core";
 
 using Azure.ClientGenerator.Core;
+using Azure.ResourceManager.CommonTypes;
+using Microsoft.DocumentDB;
 
 @@clientName(Microsoft.DocumentDB, "MongoClusterMgmtClient", "python");
+@@access(Operations.list, Access.internal, "csharp");
+@@clientName(CheckNameAvailabilityResponse,
+  "CheckNameAvailabilityResult",
+  "csharp"
+);
+@@clientName(MongoClusterRestoreParameters,
+  "MongoClusterRestoreContent",
+  "csharp"
+);
+@@clientName(CheckNameAvailabilityRequest.type, "ResourceType", "csharp");
```

Comment:
The `CheckNameAvailabilityRequest` should be renamed to `MogoClusterNameAvailabilityContent`

Reply:
And the operation should also be renamed to `CheckMongoClusterNameAvailability`